src/thor/costmatrix.cc

```diff
@@ -429,7 +452,7 @@ bool CostMatrix::ExpandInner(baldr::GraphReader& graphreader,
     // edges while still expanding on the next level since we can still transition down to
     // that level. If using a shortcut, set the shortcuts mask. Skip if this is a regular
     // edge superseded by a shortcut.
-    if (hierarchy_limits_[FORWARD][index][meta.edge_id.level() + 1].StopExpanding()) {
+    if (hierarchy_limits_[FORWARD][index][meta.edge_id.level() + 1].StopExpanding(pred.distance())) {
```

Comment:
now that we're using astar we can also disable hierarchy limits within a few kilometers of source/target to try and get higher quality routes.

Reply:
hmpf, this is a serious performance drainer though.. I just ran some tests again and all of a sudden that one test file took 100 secs instead of 60 secs.. that's a massive slowdown, also considering we didn't do that before either. I'll run a few more long-ish urban tests to see the change in route quality (taking bidir a* /route as reference).

Reply:
talked quickly with @kevinkreiser: will remove and leave a comment for future us

Reply:
fwiw now, left a comment here: https://github.com/valhalla/valhalla/pull/4671/files#diff-a2ca5ea82be964856f22db7fe58defcbe2f92e6ec7646566e5040dfef9c8229eR546

Reply:
fwiw now, left a comment here: https://github.com/valhalla/valhalla/pull/4671/files#diff-a2ca5ea82be964856f22db7fe58defcbe2f92e6ec7646566e5040dfef9c8229eR546